nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/main/java/org/apache/nifi/processors/salesforce/QuerySalesforceObject.java

```diff
@@ -162,6 +212,7 @@ public class QuerySalesforceObject extends AbstractProcessor {
             .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
             .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
             .dependsOn(AGE_FIELD)
+            .dependsOn(QUERY_TYPE, PROPERTY_BASED_QUERY)
```

Comment:
Currently CUSTOM_QUERY does not support incremental loading. That is probably intentional but in that case we should emphasize this in the `additionalDetails.html` page and maybe even in the `@CapabilityDescription`.